src/main.tf

```diff
@@ -39,13 +52,12 @@ resource "aws_eks_cluster" "cluster" {
 }
 
 resource "aws_eks_node_group" "node_group" {
-  for_each        = { for ng in var.node_groups : ng.name_suffix => ng }
-  node_group_name = "${local.cluster_name}-${each.value.name_suffix}"
-  cluster_name    = local.cluster_name
-  version         = var.k8s_version
-  subnet_ids      = local.private_subnet_ids
-  node_role_arn   = aws_iam_role.node.arn
-  instance_types  = [each.value.instance_type]
+  for_each               = { for ng in var.node_groups : ng.name_suffix => ng }
+  node_group_name_prefix = "${local.cluster_name}-${each.value.name_suffix}"
```

Comment:
This change is probably what is causing the recreation of all the nodes. Why are we switching from `node_group_name` to `node_group_name_prefix`?

Reply:
Because of node group name collision. Prior to this change, when updating the launch template and AMI, etc., I would receive an error saying the node group name already existed. By using `prefix` instead, the same node group name can be used but each one will be unique due to an added suffix.